src/Microsoft.IdentityModel.Logging/LogHelper.cs

```diff
@@ -286,8 +276,8 @@ public static Exception LogExceptionMessage(EventLevel eventLevel, Exception exc
             if (IdentityModelEventSource.Logger.IsEnabled() && IdentityModelEventSource.Logger.LogLevel >= eventLevel)
                 IdentityModelEventSource.Logger.Write(eventLevel, exception.InnerException, exception.Message);
 
-            if (Logger.IsEnabled(eventLevel))
-                Logger.Log(WriteEntry(eventLevel, exception.InnerException, exception.Message, null));
+            if (Enum.IsDefined(typeof(EventLogLevel), (int)eventLevel) && Logger.IsEnabled((EventLogLevel)eventLevel))
```

Comment:
>if (Enum.IsDefined(typeof(EventLogLevel), (int)eventLevel) && Logger.IsEnabled((EventLogLevel)eventLevel)) [](http://example.com/codeflow?start=12&length=106)

Here and below should be logging at critical or warning? Currently if the cast fails nothing is logged and flow continues, that seems incorrect to me.